### DISCUSSION TITLE
Breedbase integration with Fieldbook Question regarding trait names

### DISCUSSION THREAD

**Discussion by kristenlkoehler** [PhenoApps/Field-Book#885] [Q&A] [answered]
We are currently setting up a new Breedbase database. It integrates with fieldbook via import/export functions. 
 We would like to have a long(er) form name for traits within the database side, but use abbreviations for the fieldbook app since the screen is smaller. 
 
 Is there a way to do this? Perhaps by setting a synonym within the ontology term in the .obo file?
 
 Help and ideas are appreciated - We don't have data in the database, and I don't have an android to test with.

**Comment by trife** [comment]
@chaneylc or @bellerbrock - do you know the current behavior with synonyms?

**Reply by chaneylc** [reply]
This would be an easy change/feature, but currently not possible. The app right now checks synonym and observationVariableName, if the observationVariableName is null it will use the synonym, but you probably don't want to make that null just to make this work.

**Reply by kristenlkoehler** [reply]
Thank you very much!

**Reply by bellerbrock** [reply, marked as answer]
Sorry I'm late to the discussion, but from my experience and what [I'm seeing the the mapTraits function](https://github.com/PhenoApps/Field-Book/blob/f61bb0bc6109dc134bc70b13f69b566f3773a6ab/app/src/main/java/com/fieldbook/tracker/brapi/service/BrAPIServiceV2.java#L963) the behavior you need Kirsten is already implemented. If you include synonyms in the ontology you load into Breedbase, Field Book app will be able to access them via BrAPI and will use them preferentially over the variable names. That said we could make an improvement by having Field Book store both, or at least save the variable name in the description for reference. 

Here's an example of a trait defined in the sweetpotato ontology .obo file with multiple synonyms, and what it looks like when imported from Breedbase into Field Book via BrAPI. Field Book is just using the first synonym if there is more than one.

![synonym-import](https://github.com/PhenoApps/Field-Book/assets/11297346/7100c270-ead7-4007-b390-9aff0d2ec6bf)

**Reply by chaneylc** [reply]
This is true, I read this wrong:
```
            trait.setName(getPrioritizedValue(synonym, var.getObservationVariableName())); //This will default to the Observation Variable Name if available.
```
Assuming it would default to the obs. variable name, but it actually prioritizes the synonym field.
